src/lib/hash/hash.cpp

```diff
@@ -251,10 +251,18 @@ std::unique_ptr<HashFunction> HashFunction::create(const std::string& algo_spec,
       {
       return std::unique_ptr<HashFunction>(new SHAKE_128(req.arg_as_integer(0, 128)));
       }
+   if(req.algo_name() == "SHAKE-128(256)")
```

Comment:
This allows to create SHAKE-128/256 instances with 256/512 Bit output length by name. Maybe a more generic solution parsing the output length provided in braces is more desirable. However for the purpose of using SHAKE-128/156 covering these two cases is sufficient.